Stripe/PublicHeaders/STPRedirectContext.h

```diff
@@ -98,7 +120,35 @@ NS_EXTENSION_UNAVAILABLE("Redirect based sources are not available in extensions
  change webhooks on your backend to determine the result of a redirect.
  */
 - (nullable instancetype)initWithSource:(STPSource *)source
-                             completion:(STPRedirectContextCompletionBlock)completion;
+                             completion:(STPRedirectContextSourceCompletionBlock)completion;
+
+/**
+ Initializer for context from an `STPPaymentIntent`.
+
+ This should be used when the `status` is `STPPaymentIntentStatusRequiresSourceAction`.
+ If the next action involves a redirect, this init method will return a non-nil object.
+
+ @note You must ensure that the `returnURL` set up in the created PaymentIntent
+ correctly goes to your app so that users can be returned once
+ they complete the redirect in the web broswer. The *same* value should be passed
+ into this method, otherwise `+ [Stripe handleStripeURLCallbackWithURL:]` won't
+ recognize it.
+
+ @param paymentIntent The STPPaymentIntent that needs a redirect.
+ @param returnUrl The same `returnUrl` that the PaymentIntent has, either passed during creation
+ by your backend, or in `STPPaymentIntentParams.returnUrl`
+ @param completion A block to fire when the action is believed to have
+ been completed.
+
+ @return Nil if the provided PaymentIntent does not need a redirect. Otherwise
+ a new context object.
+
+ @note Firing of the completion block does not necessarily mean the user
```

Comment:
I prefer `Execution` to `Firing` Seems more consistent with some of the rest of your docs